google/cloud/firestore_v1/base_client.py

```diff
@@ -122,6 +123,14 @@ def __init__(
         # NOTE: This API has no use for the _http argument, but sending it
         #       will have no impact since the _http() @property only lazily
         #       creates a working HTTP object.
+        self._emulator_host = os.getenv(_FIRESTORE_EMULATOR_HOST)
+
+        if self._emulator_host is not None:
+            if credentials is None:
+                credentials = AnonymousCredentials()
+            if project is None:
+                project = "nonesuch-project-123"
```

Comment:
Default lookups for credentials / project are unnecessary (and even undesirable) when running under the emulator.

Reply:
cc77e26